students/Josh_HOff/lesson08/circle_class.py

```diff
@@ -0,0 +1,134 @@
+import sys
+import copy
+import math
+import pathlib
+from functools import total_ordering
+
+circles = []
+@total_ordering
+class Circle(object):
+
+    def __init__(self, value):
+        global circles
+        global templist
+        self._radius = value
+        self._diameter = value * 2
+        self._area = (math.pi) * (value ** 2)
+    
+    @staticmethod
+    def from_diameter(value):
+        return Circle(value/2)
+        
+    @property
+    def radius(self):
+        return self._radius
+    @radius.setter
+    def radius(self, value):
+        self._radius = value
+        self._diameter = value * 2
+        self._area = (math.pi) * (value ** 2)
```

Comment:
since you have setter/getters for diameter and area then only set them there; right now you have them duplicated in 3 places

Reply:
@natasha-aleksandrova Could you please explain this in more detail to me? Should I be removing variables or should I be changing my setter methods?

Reply:
so to simplify this is what you could have:
```
def __init__(self, radius):
    self.radius = radius

@property
def diameter(self):
    return self.radius * 2.0

@diameter.setter
def diameter(self, value):
    self.radius = value / 2.0
```
(the example is only for diameter but applies to area as well).

as I mentioned before, right now you have diameter being written in 3 places - in the snippet above the value of diameter is isolated to a single block of code, in the property setter.